modules/nextflow/src/main/groovy/nextflow/executor/BashWrapperBuilder.groovy

```diff
@@ -148,6 +152,10 @@ class BashWrapperBuilder {
         return "NXF_SCRATCH=\"\$(set +u; nxf_mktemp $scratchStr)\""
     }
 
+    protected boolean isStageFileRequired() {
+        inputFiles.size() >= 1000
```

Comment:
Worth to parametrise this with an env var, and use it to create a unit test with a few number for files